src/services/dimension-processor.ts

```diff
@@ -660,33 +668,35 @@ export const createAndValidateDateDimension = async (
 
 async function getDatePreviewWithExtractor(
   dataset: Dataset,
-  extractor: object,
   factTableColumn: string,
-  quack: Database,
-  tableName: string
+  quack: Database
 ): Promise<ViewDTO> {
-  const columnData = await quack.all(`SELECT DISTINCT "${factTableColumn}" FROM ${tableName}`);
-  const dateDimensionTable = dateDimensionReferenceTableCreator(extractor, columnData);
-  await quack.exec(createDateDimensionTable);
-  // Create the date_dimension table
-  const stmt = await quack.prepare('INSERT INTO date_dimension VALUES (?,?,?,?,?);');
-  dateDimensionTable.map(async (row) => {
-    await stmt.run(row.dateCode, row.description, row.start, row.end, row.type);
-  });
-  await stmt.finalize();
-  const countQuery = `SELECT COUNT(DISTINCT date_dimension.date_code) AS total_rows FROM date_dimension`;
-  const countResult = await quack.all(countQuery);
-  const totalRows = countResult[0].total_rows;
-
-  const previewQuery = `
-        SELECT DISTINCT(date_dimension.date_code), date_dimension.description, date_dimension.start_date, date_dimension.end_date, date_dimension.date_type
-        FROM date_dimension
-        RIGHT JOIN "${tableName}" ON CAST("${tableName}"."${factTableColumn}" AS VARCHAR)=CAST(date_dimension.date_code AS VARCHAR)
+  const tableName = `${makeCubeSafeString(factTableColumn)}_lookup`;
+  const totalsQuery = await quack.all(
+    pgformat('SELECT COUNT(DISTINCT %I) AS totalLines FROM %I;', factTableColumn, 'fact_table')
+  );
+  const previewQuery = pgformat(
+    `
+        SELECT DISTINCT(%I.%I), %I.description, %I.start_date, %I.end_date, %I.date_type
+        FROM %I.%I
+        RIGHT JOIN fact_table ON CAST(fact_table.%I AS VARCHAR)=CAST(%I.%I AS VARCHAR)
         ORDER BY end_date ASC
-        LIMIT ${sampleSize}
-    `;
+        LIMIT %L
+    `,
+    tableName,
+    factTableColumn,
+    tableName,
+    tableName,
+    tableName,
+    tableName,
+    dataset.draftRevision!.id,
+    tableName,
+    factTableColumn,
+    tableName,
+    factTableColumn,
+    sampleSize
```

Comment:
This feels a bit awkward to use and possibly prone to mistakes - we might want to think about creating a helper function that maps a single copy of each param to the necessary string format for `pgformat`. Not a blocker for this PR though.